plugins/image-prioritizer/class-image-prioritizer-img-tag-visitor.php

```diff
@@ -126,4 +157,20 @@ static function ( string $value ): bool {
 
 		return true;
 	}
+
+	/**
+	 * Checks whether the given 'sizes' attribute includes the 'auto' keyword as the first item in the list.
+	 *
+	 * Per the HTML spec, if present it must be the first entry.
+	 *
+	 * @since n.e.x.t
+	 * @see auto_sizes_attribute_includes_valid_auto()
+	 *
+	 * @param string $sizes_attr The 'sizes' attribute value.
+	 * @return bool True if the 'auto' keyword is present, false otherwise.
+	 */
+	private function sizes_attribute_includes_valid_auto( string $sizes_attr ): bool {
+		list( $first_size ) = explode( ',', $sizes_attr, 2 );
+		return 'auto' === strtolower( trim( $first_size, " \t\f\r\n" ) );
+	}
```

Comment:
See above, I think this should be removed. It's too much only intended for edge cases to have this logic duplicated across three different codebases.
```suggestion
```